rocky/rocky/scheduler.py

```diff
@@ -215,10 +215,12 @@ def get_lazy_task_list(
             boefje_name=boefje_name,
         )
 
-    def get_task_details(self, task_id) -> Task:
+    def get_task_details(self, organization_code: str, task_id: str) -> Optional[Task]:
         res = self.session.get(f"{self._base_uri}/tasks/{task_id}")
         res.raise_for_status()
-        return Task.model_validate_json(res.content)
+        task_details = Task.model_validate_json(res.content)
+        if task_details.p_item.data.organization == organization_code:
+            return task_details
```

Comment:
Should ideally be fixed in the scheduler API

Reply:
Agree!

Reply:
@jpbruinsslot Is this something you can implement for me?

Reply:
@ammar92 what specifically is expected from the scheduler api? Since permission checking of resources is afaik the responsibility of rocky.

Reply:
> @ammar92 what specifically is expected from the scheduler api? Since permission checking of resources is afaik the responsibility of rocky.

To be able to filter on tasks (e.g. by querying for an organisation id next to the task id). This would mean less filtering on the frontend

Reply:
This is possible in the scheduler api [filtering](https://github.com/minvws/nl-kat-coordination/blob/main/mula/docs/api.md#filtering). @Rieven let me know if you want me to walk you through it

Reply:
I am going to refactor this part in another PR